src/components/TradeWidget/index.tsx

```diff
@@ -461,8 +461,8 @@ const TradeWidget: React.FC = () => {
         dispatch(removePendingOrdersAction({ networkId, pendingTxHash, userAddress }))
       }
     } else {
-      const from = history.location.pathname + history.location.search
-      history.push('/connect-wallet', { from })
+      // Not connected. Prompt user to connect his wallet
+      await connectWallet()
```

Comment:
So it only asks to connect, but doesn't submit the transaction after connection
Kinda not intuitive

Reply:
this is where redux middleware would be very useful

Reply:
can just send the tx if connection is successful

Reply:
Would you rather do that after connecting?

Reply:
```js
if (!isConnected) //send tx
else {
  // connect
  if (success) // send tx
}
```

would be my preference
@anxolin , @W3stside, @biocom  ?

Reply:
yes i think it makes the most sense :100: